src/commands/channels/configure.ts

```diff
@@ -0,0 +1,140 @@
+/*
+* Copyright 2022, Nuance, Inc. and its contributors.
+* All rights reserved.
+*
+* This source code is licensed under the Apache-2.0 license found in
+* the LICENSE file in the root directory of this source tree.
+*/
+
+import {flags} from '@oclif/command'
+import chalk from 'chalk'
+import makeDebug from 'debug'
+
+import * as ChannelsAPI from '../../mix/api/channels'
+import {ChannelModalities, ChannelsConfigParams} from '../../mix/api/channels-types'
+import {channelColors} from '../../mix/api/utils/channel-colors'
+import {MixClient, MixResponse, MixResult} from '../../mix/types'
+import {asArray} from '../../utils/as-array'
+import MixCommand from '../../utils/base/mix-command'
+import * as MixFlags from '../../utils/flags'
+import {DomainOption} from '../../utils/validations'
+
+const debug = makeDebug('mix:commands:channels:configure')
+
+export default class ChannelsConfigure extends MixCommand {
+  static description = `update channel details in a project
+
+  Configure the modalities and color of an existing channel in a Mix project.
+  `
+
+  static examples = [
+    'mix channels:configure -P 1922  \\',
+    '  --channel bc40667c-e0f6-11ec-9d64-0242ac120003 \\',
+    '  --mode DTMF --mode TTS \\',
+    '  --color SALMON',
+  ]
+
+  static flags = {
+    project: MixFlags.projectFlag,
+    channel: flags.string({
+      char: 'C',
+      description: 'channel id',
+      required: true,
+    }),
+    mode: {
+      ...MixFlags.modesFlag,
+      required: false,
+    },
+    color: flags.string({
+      description: 'channel color',
+    }),
+    json: MixFlags.jsonFlag,
+    yaml: MixFlags.yamlFlag,
+  }
+
+  get domainOptions(): DomainOption[] {
+    debug('get domainOptions()')
+    return ['project']
+  }
+
+  tryDomainOptionsValidation(options: any, domainOptions: DomainOption[]) {
+    debug('tryDomainOptionsValidation()')
+    super.tryDomainOptionsValidation(options, domainOptions)
+
+    // At least one flag must be set
+    if (!options.mode && !options.color) {
+      this.error('At least one of --mode and --color must be set.')
+    }
+
+    const color = options.color?.toUpperCase()
+
+    if (color && !channelColors.slice(1).includes(color)) {
+      this.error(`Unknown color ${chalk.red(options.color)}.`,
+        {suggestions: ['check value supplied to --color flag and try again.']})
+    }
+
+    const modesSeen = Object.fromEntries(
+      ChannelModalities
+        .slice(1)
+        .map((mode: string) => [mode, false]))
+
+    const modesOptions = options.mode?.map((mode: string) => mode.toUpperCase())
+
+    for (const mode of modesOptions ?? []) {
+      if (!(mode in modesSeen)) {
+        // TODO
+        this.error(`Unknown mode ${chalk.red(mode)}`)
+      } else if (modesSeen[mode]) {
+        // TODO
+        this.error(`Duplicate mode ${chalk.red(mode)} appears twice.`)
```

Comment:
Perhaps use a message that does not commit to how many times it's been seen, e.g. "Mode <mode> specified more than once". I will let you decide on the wording :) but I feel "duplicate" and "twice" sound redundant and it could be more than twice.